src/bcm2835.c

```diff
@@ -933,6 +943,46 @@ void bcm2835_spi_write(uint16_t data)
 #endif
 }
 
+void bcm2835_pde_set_separator_duration(uint32_t pin, uint32_t duration)
+{
+	if (!pin_settings_map.contains(pin)) {
+		pin_settings_map.emplace(pin, {});
+	}
+
+	pin_settings_map.at(pin).separatorDuration = duration;
+}
+
+void bcm2835_pde_set_short_duration(uint32_t pin, uint32_t duration)
+{
+	if (!pin_settings_map.contains(pin)) {
+		pin_settings_map.emplace(pin, {});
+	}
+
+	pin_settings_map.at(pin).shortDuration = duration;
+}
+
+void bcm2835_pde_set_long_duration(uint32_t pin, uint32_t duration)
+{
+	if (!pin_settings_map.contains(pin)) {
+		pin_settings_map.emplace(pin, {});
+	}
+
+	pin_settings_map.at(pin).longDuration = duration;
+}
+
+void bcm2835_pde_set_separator(uint32_t pin, uint32_t separator)
+{
+	if (!pin_settings_map.contains(pin)) {
+		pin_settings_map.emplace(pin, {});
+	}
+
+	pin_settings_map.at(pin).separator = separator;
+}
```

Comment:
Since these are identical in bcm2835 and sunxi, can they be moved to rpio.cc and just used from there?